lib/asyn-thread.c

```diff
@@ -454,6 +626,37 @@ static bool init_resolve_thread(struct Curl_easy *data,
   /* The thread will set this to 1 when complete. */
   td->tsd.done = 0;
 
+#ifdef _WIN32
+  if(Curl_isWindows8OrGreater && Curl_FreeAddrInfoExW &&
+      Curl_GetAddrInfoExCancel && Curl_GetAddrInfoExW) {
+    int hostname_w_len = MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS,
+        hostname, -1, NULL, 0); /* calculate required length */
```

Comment:
BTW, this host name is known to be ascii here. Does that change anything? Do you still need to do this funny wide character dance?

If the name was provided an an IDN, it was already converted before this and the host name is the punycode (all ASCII) version.

Reply:
I could simply loop there instead.

Reply:
If the name was not IDN though wouldn't it pass through here? It seems safer to leave it